portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,20 +14,96 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.ArrayList;
 
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private final static String tblTitle = "Comment";
+  private final static String tblComment = "comment";
+  private final static String tblName = "name";
+  private final static String tblTime = "timestamp";
+
+  private final static String htmlComment = "text-input";
+  private final static String htmlName = "name";
+  private final static String htmlLimit = "commentLimit";
+
+  private static int limit = -1;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Jacob!</h1>");
+    Query query = new Query(tblTitle).addSort(tblTime, SortDirection.DESCENDING);;
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<Comment>();
+    int count = 0;
+    for (Entity entity : results.asIterable()) {
+      if (limit > 0) {
+        if(count < limit){
+        String commentString = (String) entity.getProperty(tblComment);
+        String name = (String) entity.getProperty(tblName);
+        long id = entity.getKey().getId();
+
+        Comment comment = new Comment(id, commentString, name);
+        comments.add(comment);
+        count++;
+        }else {
+          break;
+        }
+      }else {
+        String commentString = (String) entity.getProperty(tblComment);
+        String name = (String) entity.getProperty(tblName);
+        long id = entity.getKey().getId();
+
+        Comment comment = new Comment(id, commentString, name);
+        comments.add(comment);
+      }
+    }
+
+    Gson gson = new Gson();
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
   }
-}
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = request.getParameter(htmlComment);
+    String name = request.getParameter(htmlName);
+    long timestamp = System.currentTimeMillis();
+
+    // Add input to current comments in datastore.
+    Entity taskEntity = new Entity(tblTitle);
+    taskEntity.setProperty(tblComment, text);
+    taskEntity.setProperty(tblName, name);
+    taskEntity.setProperty(tblTime, timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
+
+     // Set how many comments will be printed after redirect.
+     limit = Integer.parseInt(request.getParameter(htmlLimit));
```

Comment:
I really don't like the fact of limit is set via "Post" request but it's only used in "Get" request.

Reply:
Well my "Post" request redirects to index.html triggering "Get", and the number of comments is only specified during the submit. The first time the page opens it displays all the comments. 

I will still see if there is an alternate way for me to access this value in my "Get". The problem I previously faced was that request.getParameter(htmlLimit) returned null.

Reply:
My comment above explains how I implemented your suggestion.